src/util/request-manager.js

```diff
@@ -503,16 +512,28 @@ export default class RequestManager {
     const process = params.process;
     if (process) {
       req.on('response', res => {
+        if (res.statusCode === 408 || res.statusCode >= 500) {
+          // Retryable server responses
+          const description = `${res.statusCode} ${http.STATUS_CODES[res.statusCode]}`;
+          if (!queueForRetry(this.reporter.lang('internalServerErrorRetrying', description))) {
```

Comment:
Based on the usage here and above, it feels to me that `queueForRetry` should either be an async function, raising the exception itself or take in a `reject` callback and do the rejection itself.